File: Dart/lib/findPermutationDifference.dart
class Solution {
  int findPermutationDifference(String s, String t) {
    var count = List<int>.filled(26, 0);
    var sum = 0;
    for (var i = 0; i < s.length; i++) {
      count[s.codeUnitAt(i) - 'a'.codeUnitAt(0)] = i;
    }
    for (var i = 0; i < t.length; i++) {
      sum += (count[t.codeUnitAt(i) - 'a'.codeUnitAt(0)] - i).abs();
    }
    return sum;
  }
}

File: Ruby/solution/find_permutation_difference.rb
# @param {String} s
# @param {String} t
# @return {Integer}
def find_permutation_difference(s, t)
  s_map = Hash.new(0)
  sum = 0
  s.chars.each_with_index { |char, i| s_map[char] = i }
  t.chars.each_with_index { |char, i| sum += (s_map[char] - i).abs }
  sum
end
